test/core/test/__snapshots__/jest-expect.test.ts.snap

```diff
@@ -231,35 +231,69 @@ exports[`asymmetric matcher error 16`] = `
 exports[`asymmetric matcher error 17`] = `
 {
   "actual": "hello",
-  "diff": undefined,
+  "diff": null,
   "expected": "StringContaining "xx"",
-  "message": "expected thrown value 'hello' to match asymmetric matcher: StringContaining "xx"",
+  "message": "expected error to match asymmetric matcher",
```

Comment:
Hmm, I just tested `showDiff: true`, but diff is not showing up for this case:

```tsx
expect(() => { throw "hello" }).toThrow(expect.stringContaining("xx"))
```

Reply:
It like this (odd) case leads to `omitDifference`  here

https://github.com/vitest-dev/vitest/blob/9d3a7de94c83c862fc751b2aba7915e1c9ef7cd4/packages/utils/src/diff/index.ts#L85-L88